samza-core/src/test/java/org/apache/samza/zk/TestZkMetadataStore.java

```diff
@@ -41,7 +41,13 @@ public class TestZkMetadataStore {
 
   private static final Random RANDOM = new Random();
 
-  private static final int VALUE_SIZE_IN_BYTES = 1024  * 1024 * 10; // 10 MB
+  // In 101tec version, this constant was 1024 * 1024 * 10
+  // The limit changed to 1024 * 1000 after helix migration.
+  // A larger number would throw an exception.
+  // See: https://github.com/apache/helix/blob/654636e54268907deb2e12d32913455cc543b436/zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java#L2386
+  // The limit can be set through a system property, but not in unit tests.
+  // https://github.com/apache/helix/blob/654636e54268907deb2e12d32913455cc543b436/zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java#L106
+  private static final int VALUE_SIZE_IN_BYTES = 512000;
```

Comment:
Was that change to 1024*1000 inside LinkedIn? If so, we should keep this `1024*1024*10` in OSS if helix client accepts this value.

Reply:
The change was not inside LinkedIn, though they provide a way to override it. This is probably the most concerning change on this PR.

Reply:
Since its used within the tests, i want to check if we are losing on the test coverage of testing the higher scale of the client (if it was intentional)

Reply:
Why is it concerning? Isn't it used for test only?

Reply:
We don't know if actual clients need this size.

Reply:
@mynameborat what's the path forward here? Can we use version bumps?

Reply:
We may need to just validate this internally before rolling this out. Maybe add a note in the PR description about potential degradation as this isn't tested.